includes/admin/class-wc-rest-upe-flag-toggle-controller.php

```diff
@@ -87,7 +87,7 @@ public function set_flag( WP_REST_Request $request ) {
 		}
 
 		$settings = get_option( 'woocommerce_stripe_settings', [] );
-		$settings[ WC_Stripe_Feature_Flags::UPE_CHECKOUT_FEATURE_ATTRIBUTE_NAME ] = $is_upe_enabled ? 'yes' : 'no';
+		$settings[ WC_Stripe_Feature_Flags::UPE_CHECKOUT_FEATURE_ATTRIBUTE_NAME ] = $is_upe_enabled ? 'yes' : 'disabled';
```

Comment:
I wanted to avoid creating a new entry in `wp_options` to track whether the feature had been manually disabled to hide the note/onboarding task.

Reply:
Are we going to use the `did_merchant_disable_upe` function later for hiding other features also? 

If not then deleting the note could be another solution here (delete it when the POST request has `is_upe_enabled` param false indicating a disable request through UI)

I have noticed that once a notice is dismissed from the inbox it does not appear again.

Reply:
> Are we going to use the `did_merchant_disable_upe` function later for hiding other features also? 

At this time I could only think of this situation where `did_merchant_disable_upe` is needed

> I have noticed that once a notice is dismissed from the inbox it does not appear again.

That is correct, once deleted with `possibly_delete_note`, the note disappears (I have [this](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1841) that takes care of that scenario).
The issue here is that if the merchant disables UPE, the note would appear again. With this PR, the note would not appear anymore, even after disabling UPE.